src/hipscat_import/index/map_reduce.py

```diff
@@ -0,0 +1,43 @@
+"""Create columnar index of hipscat table using dask for parallelization"""
+
+import dask.dataframe as dd
+import numpy as np
+from hipscat.io import file_io
+
+
+def create_index(args):
+    """Read primary column, indexing column, and other payload data,
+    and write to catalog directory."""
+    include_columns = [args.indexing_column]
+    if args.extra_columns:
+        include_columns.extend(args.extra_columns)
+    if args.include_order_pixel:
+        include_columns.extend(["Norder", "Dir", "Npix"])
+
+    index_dir = file_io.append_paths_to_pointer(args.catalog_path, "index")
+
+    data = dd.read_parquet(
+        path=args.input_catalog_path,
+        columns=include_columns,
+        engine="pyarrow",
+        dataset={"partitioning": "hive"},
+    )
```

Comment:
Yes - but it might be better to add something like `hipscat.io.is_valid_catalog` that will check for file/directory existence. It would also be nice to have in the association pipeline.

Created https://github.com/astronomy-commons/hipscat/issues/88

Reply:
Awesome, thanks!